src/ccnl-core/src/ccnl-prefix.c

```diff
@@ -595,29 +595,29 @@ ccnl_prefix_to_str_detailed(struct ccnl_prefix_s *pr, int ccntlv_skip, int escap
     (void)j;
     (void)len;
     (void) call_slash;
-    (void)buflen; //fixme, prevent bufferoverrun here!
+    (void)buflen;
 #ifdef USE_NFN
-    // len += sprintf(buf + len, "cmpcnt: %i|", pr->compcnt);
+    // len += snprintf(buf + len, buflen - len, "cmpcnt: %i|", pr->compcnt);
     if (pr->nfnflags & CCNL_PREFIX_NFN) {
-        len += sprintf(buf + len, "nfn");
+        len += snprintf(buf + len, buflen - len, "nfn");
     }
 #ifdef USE_NFN_REQUESTS
-    // len += sprintf(buf + len, ":test");
+    // len += snprintf(buf + len, buflen - len, ":test");
     if (pr->nfnflags & CCNL_PREFIX_REQUEST) {
         char *desc = nfn_request_description_new(pr->request);
-        len += sprintf(buf + len, ":%s", desc);
+        len += snprintf(buf + len, buflen - len, ":%s", desc);
```

Comment:
The good old boy scout rule, i.e. you should check the return value of snprintf. This applies to all usage of snprintf in this PR.

Reply:
It's not only to improve the code quality here, I just found out, that the bug is not fixed, if the return value is not checked, thanks for the hint

Reply:
I fixed that